libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/exception.kt

```diff
@@ -5,23 +5,23 @@ import java.io.IOException
 
 open class SourceAwareException(
     val error: String,
-    val sourceLocation: SourceLocation,
+    val sourceLocation: SourceLocation?,
 ) : RuntimeException(preview(
     error = error,
     sourceLocation = sourceLocation
 )) {
 
   companion object {
-    private fun formatForIdea(sourceLocation: SourceLocation, description: String): String {
+    private fun formatForIdea(sourceLocation: SourceLocation?, description: String): String {
       // Idea understands a certain format and makes logs clickable
       // It's not 100% clear where this is specified but it at least works with
       // 2020.3
-      return "e: ${sourceLocation.pretty()}: $description"
+      return "e: ${sourceLocation?.pretty()}: $description"
```

Comment:
We could have a `fun SourceLocation?.pretty()` that returns `"(unknown location)"`?

Reply:
Good idea, added [here](https://github.com/apollographql/apollo-kotlin/pull/5064/commits/d041d48f43266cab7270686142298745e287b8a7)